framework/src/materials/ParsedMaterialHelper.C

```diff
@@ -47,8 +47,8 @@ ParsedMaterialHelper<is_ad>::functionParse(const std::string & function_expressi
                                            const std::vector<std::string> & constant_names,
                                            const std::vector<std::string> & constant_expressions)
 {
-  std::vector<std::string> empty_string_vector;
-  std::vector<Real> empty_real_vector;
+  static const std::vector<std::string> empty_string_vector;
```

Comment:
why static? Is there going to be threaded initialization of this vector?
What about access?

Reply:
Static cause the vector to be built _once_. I'm pretty confident that this function is not hit in a multithreaded region (for example I'm copying derivatives between threads, so that the AD is only run once, which happens during object construction, which is not threaded).
But it might be better more defensive coding to make this not const (as this is unlikely to be a performance issue anyways)